### PR TITLE
spec: add spec for notation version command

### PR DESCRIPTION
**Pull request by yizha1**
Add spec for `notation version` command.

### CODE REVIEW

File: specs/commandline/version.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+Use `notation version` to print notation version information.
+
+Upon successful execution, the following information is printed.
+
+```text
+Notation: Notary v2, A tool to sign, store, and verify artifacts.
+
+Version:    <version_in_format_vX.Y.Z[suffix]>
+Go version: <version in format goX.Y.Z>
+Git commit: <commit_id_in_7_characters>
```
Comment (priteshbandi) [inline specs/commandline/version.md:14]: Why do we need to display go version and git commit? Can they be inferred from `Version`?

Comment (dtzar) [reply specs/commandline/version.md:14]: I agree - we should have just the basic version there but if someone wants more, they could perhaps do the -v (verbose) to get additional information.  This is what you get with **kubectl version** for instance:

```
$kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.2", GitCommit:"5835544ca568b757a8ecae5c153f317e5736700e", GitTreeState:"clean", BuildDate:"2022-09-21T14:33:49Z", GoVersion:"go1.19.1", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.7

$kubectl version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.25.2
Kustomize Version: v4.5.7
```

**Review by dtzar — CHANGES_REQUESTED** [review]

**Review by dtzar — APPROVED** [review]

File: specs/commandline/version.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+## Description
+
+Use `notation version` to print notation version information.
+
+Upon successful execution, the following information is printed.
+
+```text
+Notation: Notary v2, A tool to sign, store, and verify artifacts.
+
+Version:    <version_in_format_vX.Y.Z[suffix]>
```
Comment (priteshbandi) [inline specs/commandline/version.md:12]:
```suggestion
Version:    <MAJOR.MINOR.PATCH>
```

I would omit suffix as we are doing this for RC1

**Review by priteshbandi — APPROVED** [review]
Added one feedback. Please incorporate before pushing change.

File: specs/commandline/version.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+```shell
+notation version
+```
+
+An example output:
+
+```text
+Notation: Notary v2, A tool to sign, store, and verify artifacts.
+
+Version:    1.0.0
```
Comment (shizhMSFT) [inline specs/commandline/version.md:40]: We also need to show the following metadata with justifications
- Go version: Different go version has different built-in library implementation, including crypto implementation. Note that the same notation version can be compiled with different golang versions.
- Git Commit & Git tree state: Users / downstream may build by their own. We have to know if they are using the right commit with clean tree state.

Comment (shizhMSFT) [reply specs/commandline/version.md:40]: For instances,

oras downloaded from github release page:
```console
$ oras version
Version:        0.15.1
Go version:     go1.19
Git commit:     36dc01d02c54bf410d56d5e16b597eedbe666ec6
Git tree state: clean
```

oras downloaded from homebrew:
```console
$ oras version
Version:    0.15.1+Homebrew
Go version: go1.19.2
```

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

File: specs/commandline/version.md (hunk 3 above)
Comment (yizha1) [reply specs/commandline/version.md:40]: This comment is valid.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**